generators/vue/templates/src/main/webapp/app/account/account.service.ts.ejs

```diff
@@ -15,78 +13,68 @@ export default class AccountService {
-        this.store.commit('authenticate');
+  public async retrieveAccount(): Promise<boolean> {
+    try {
+      const response = await axios.get<any>('api/account');
+      if (response.status === 200 && response.data) {
         const account = response.data;
-        if (account) {
-          this.store.commit('authenticated', account);
+        this.store.commit('authenticated', account);
 <%_ if (enableTranslation) { _%>
-          if (this.store.getters.currentLanguage !== account.langKey) {
-            this.store.commit('currentLanguage', account.langKey);
-          }
-<%_ } _%>
-          if(sessionStorage.getItem('requested-url')) {
-            this.router.replace(sessionStorage.getItem("requested-url"));
-            sessionStorage.removeItem("requested-url");
-          }
-        } else {
-          this.store.commit('logout');
-          if (this.router.currentRoute.path !== '/') {
-            this.router.push('/');
-          }
-          sessionStorage.removeItem("requested-url");
+        if (this.store.getters.currentLanguage !== account.langKey) {
+          this.store.commit('currentLanguage', account.langKey);
+          this.translationService.refreshTranslation(this.store.getters.currentLanguage);
         }
-<%_ if (enableTranslation) { _%>
-        this.translationService.refreshTranslation(this.store.getters.currentLanguage);
 <%_ } _%>
-        resolve(true);
-      }).catch(() => {
-        this.store.commit('logout');
-        resolve(false);
-      });
-    });
+        return true;
+      }
+    } catch (error) {}
+
+    this.store.commit('logout');
+    return false;
   }
 
-  public hasAnyAuthorityAndCheckAuth(authorities: any): Promise<boolean> {
-    if (typeof authorities === 'string') {
-      authorities = [authorities];
+  public async loadAccount() {
+    if (this.store.getters.logon) {
+      return this.store.getters.logon;
     }
 
-    if (!this.authenticated || !this.userAuthorities) {
-      const token = <%_ if (authenticationTypeJwt) { _%> localStorage.getItem('<%=jhiPrefixDashed %>-authenticationToken') || sessionStorage.getItem('<%=jhiPrefixDashed %>-authenticationToken'); <%_ } else { _%> this.cookie.get('JSESSIONID') || this.cookie.get('XSRF-TOKEN'); <%_ } _%>
-      if (!this.store.getters.account && !this.store.getters.logon && token) {
-        return this.retrieveAccount().then(resp => {
-          if (resp) {
-            return this.checkAuthorities(authorities);
-          }
-          return Promise.resolve(false);
-        });
-      }
-      return Promise.resolve(false);
+    const token = localStorage.getItem('<%=jhiPrefixDashed %>-authenticationToken') || sessionStorage.getItem('<%=jhiPrefixDashed %>-authenticationToken');
+    if (this.authenticated && this.userAuthorities && token) {
+      return;
+    }
+
+    const promise = this.retrieveAccount();
+    this.store.commit('authenticate', promise);
+    return await promise;
```

Comment:
https://eslint.org/docs/latest/rules/no-return-await I think it's not a good practice

Reply:
Leftover thanks..